.circleci/config.yml

```diff
@@ -0,0 +1,302 @@
+  app-build-and-store:
+    description: Run build and store archive of it
+    steps:
+      - setup-build-workspace
+      - setup-shared-env
+      - build-app
+      - store-app-build
+
+  manager-build-and-deploy:
+    description: Run build and immediately deploy it to GAE
+    steps:
+      - setup-build-workspace
+      - setup-shared-env
+      - build-app
+      - deploy-exploded-build
+
+  set-deployment-environment:
+    description: "Determine working environment"
+    steps:
+      - run:
+          name: Determine environment for branch << pipeline.git.branch >>
+          command: |
+            case "<< pipeline.git.branch >>" in
+              develop)
+                DEPLOY_ENV=dev
+              ;;
+              angular-appengine-deployment-ctd-2)
+                DEPLOY_ENV=dev
+              ;;
+              test)
+                DEPLOY_ENV=test
+              ;;
+              staging)
+                DEPLOY_ENV=staging
+              ;;
+              production)
+                DEPLOY_ENV=prod
+              ;;
+              *)
+                echo "Cannot map << pipeline.git.branch >> to a deployment environment"
+                exit 17
+              ;;
+            esac
+            echo "Setting deployment ENVIRONMENT to $DEPLOY_ENV"
+            echo "export ENVIRONMENT=$DEPLOY_ENV" >> $BASH_ENV
+            source $BASH_ENV
+
+  deploy-stored-build:
+    description: Deploy archived build to GAE
+    steps:
+      - checkout:
+          path: ~/repo
+      - setup-shared-env
+      - set-deployment-environment
+      - run:
+          name: Retrieve and expand build archive for git SHA << pipeline.git.revision >>
+          command: |
+            set -u
+            # we are getting our builds from the one bucket
+            readvault.sh secret/pepper/prod/v1/conf .data.gcp.serviceKey | gcloud auth activate-service-account --key-file=-
+            TAR_FILE_URL_PATTERN="gs://${BUILDS_BUCKET}/study-manager_*_${SHORT_GIT_SHA}.tar.gz"
+            TAR_FILE_URL=`gsutil ls $TAR_FILE_URL_PATTERN  | sort -r | head -1`
+
+            if [ -z $TAR_FILE_URL ]
+            then
+              echo "Could not find archive using pattern ${TAR_FILE_URL_PATTERN}"
+              exit 1
+            fi
+            TAR_FILE_PATH="/tmp/$(basename -- $TAR_FILE_URL)"
+            gsutil cp  $TAR_FILE_URL $TAR_FILE_PATH
+            cd ~/repo/client
+            mkdir -p $OUTPUT_DIR && tar -xvf $TAR_FILE_PATH -C $OUTPUT_DIR
+            echo "Contents of extracted archive at $OUTPUT_DIR"
+            ls -l $OUTPUT_DIR
+      - deploy-exploded-build
+
+  deploy-exploded-build:
+    description: Deploy files to GAE
+    steps:
+      - set-deployment-environment
+      - run:
+          name: generate config files and copy pepperConfig.js
+          command: |
+            set -u
+            ./build-study.sh v1 $ENVIRONMENT . --config
+            cd ddp-workspace
+            CONFIG_DIR="${OUTPUT_DIR}/assets/config"
```

Comment:
Not sure where generated values for js should go in study-manager, so this will probably need some changes.